app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java

```diff
@@ -185,6 +185,8 @@ public void onReceive(Context context, Intent intent) {
     // Experimental support for rfduino from Tomasz Stachowicz
     private static volatile BluetoothGattCharacteristic mCharacteristicSend;
     private byte[] lastdata = null;
+    private static BluetoothManager myBluetoothManager = null; // for display in system status
```

Comment:
I'm not really sure this should be being held in a static field - or indeed at all. I would imagine the static reference could create issues with garbage collection as its a system service. Can't the instance of this just be retrieved where you are displaying connected devices in engineering mode megaStatus() ?

Reply:
But problem that I need to call getSystemService() which cant be called from within a static function.
Ýou have more knowledge than me so I simply removed it.

Reply:
You can just do ` xdrip.getAppContext().getSystemService()` from a static context.

Reply:
oh thanks for hints, I changed code accordingly and squashed